main.py

```diff
@@ -1,40 +1,55 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 """A Random Walk Simulation """
-
-# Python code for 2D random walk.
-# Source: https://www.geeksforgeeks.org/random-walk-implementation-python/
-import numpy
+import argparse
 import matplotlib.pyplot as plt
-import random
+import random_walker
+from playground import Playground
 
-# defining the number of steps
-n = 100000
 
-# creating two array for containing x and y coordinate
-# of size equals to the number of size and filled up with 0's
-x = numpy.zeros(n)
-y = numpy.zeros(n)
+def main(steps: int, walkers: int, save: bool, map_seed: int, land_scale: int):
+    """
+    Execute the random walker simulation for a given set of parameters
 
-# filling the coordinates with random variables
-for i in range(1, n):
-    val = random.randint(1, 4)
-    if val == 1:
-        x[i] = x[i - 1] + 1
-        y[i] = y[i - 1]
-    elif val == 2:
-        x[i] = x[i - 1] - 1
-        y[i] = y[i - 1]
-    elif val == 3:
-        x[i] = x[i - 1]
-        y[i] = y[i - 1] + 1
-    else:
-        x[i] = x[i - 1]
-        y[i] = y[i - 1] - 1
+    :param steps: number of steps per walker
+    :param walkers: number of walkers
+    :param save: save the created figure
+    :param map_seed: map seed for the playground
+    :param land_scale: scale for the playground
+    :return:
+    """
+    # Create Playground and walkers
+    playground = Playground(seed=map_seed, scaling=land_scale)
+    walker_list = random_walker.create_different_walkers(walkers, steps)
+    # Add Playground to plt
+    plt.title("Random Walk ($n = " + str(steps) + "$ steps)")
+    for x_positions, y_positions in playground.get_line_segments():
+        plt.plot(x_positions, y_positions, color='red')
+    # for each walker calculate the random walk and add to plt
+    for walker_index in range(walkers):
+        walker = walker_list[walker_index]
+        walker.execute_random_walk(playground)
+        # plotting the walk
+        plt.plot(walker.x_positions,
+                 walker.y_positions,
+                 label=str(type(walker).__name__) + ' index: ' + str(walker_index))
+    # optional save the plot
+    if save:
```

Comment:
![image](https://user-images.githubusercontent.com/73953672/128150847-df291931-552a-44ac-a9b6-39211348a9b7.png)
python mian.py -w 3 -ls 200 -ps 2 -s 100
that is not how it should look like I guess? or is it?